docs/jdbc.md

```diff
@@ -0,0 +1,456 @@
+Properties connProps=new Properties();
+connProps.setProperty("user","<USER_NAME>");
+connProps.setProperty("password","<PASSWORD>");
+connProps.setProperty("encrypt","false");
+connProps.setProperty("socketFactory","com.google.cloud.sql.sqlserver.SocketFactory");
+
+connProps.setProperty("socketFactoryConstructorArg",
+"<INSTANCE_CONNECTION_NAME>?ipTypes=PRIVATE");
+
+// Initialize connection pool
+HikariConfig config=new HikariConfig();
+config.setJdbcUrl(jdbcURL);
+config.setDataSourceClassName("com.microsoft.sqlserver.jdbc.SQLServerDataSource");
+config.setDataSourceProperties(connProps);
+config.setConnectionTimeout(10000); // 10s
+
+HikariDataSource connectionPool=new HikariDataSource(config);
+```
+
+### IAM Authentication
+
+**Note:** This feature is currently only supported for Mysql and Postgres
+drivers.
+
+Connections using
+[IAM database authentication](https://cloud.google.com/sql/docs/postgres/iam-logins)
+are supported when connecting to Mysql or Postgres instances.
+This feature is unsupported for SQL Server. First, make sure to
+[configure your Cloud SQL Instance to allow IAM authentication](https://cloud.google.com/sql/docs/postgres/create-edit-iam-instances#configure-iam-db-instance)
+and
+[add an IAM database user](https://cloud.google.com/sql/docs/postgres/create-manage-iam-users#creating-a-database-user).
+Now, you can connect using user or service
+account credentials instead of a password.
+When setting up the connection, set the `ENABLE_IAM_AUTH` connection property
+to `true` and `user`
+to the email address associated with your IAM user.
+
+You must shorten the full IAM user email into a database username. Due to
+different constraints on allowed characters in the database username, Mysql and
+postgres differ in how they shorten an IAM email address into a database
+username.
+
+* Mysql: Truncate the IAM email removing the `@` and everything that follows.
+* Postgres: If the IAM email ends with `.gserviceaccount.com`, remove
+  the `.gserviceaccount.com` suffix from the email.
+
+For example, if the full IAM user account is
+`my-sa@my-project.iam.gserviceaccount.com`, then the shortened database username
+would be `my-sa` for Mysql, and `my-sa@my-project.iam` for Postgres.
+
+**Note:** a non-empty string value for the `password` property must be set.
+While this property will be ignored when connecting with the Cloud SQL Connector
+using IAM auth, leaving it empty will cause driver-level validations to fail.
+
+#### Example
+
+Replace these parameters in the example based on your database type:
+
+|          | JDBC_URL                              | DRIVER_CLASS                                | IAM_DB_USER          |
+|----------|---------------------------------------|---------------------------------------------|----------------------|
+| Mysql    | jdbc:mysql:///<DB_NAME>               | com.google.cloud.sql.mysql.SocketFactory    | my-sa                |  
```

Comment:
Maybe we don't need this first column either?

Reply:
I like it for clarity.